core/frontend/src/store/settings.ts

```diff
@@ -93,7 +93,7 @@ class SettingsStore extends VuexModule {
      */
     static saveVariable<T>(name: string, value: T): void {
       // eslint-disable-next-line
-      window.localStorage.setItem(SettingsStore.settingsName(name), value as any)
+      window.localStorage.setItem(SettingsStore.settingsName(name), JSON.stringify(value))
```

Comment:
probably `value` here needs to have no hidden fields to be retrieved with no errors from the store... So why not change the type T to JSON to avoid the trap?

Reply:
> probably `value` here needs to have no hidden fields to be retrieved with no errors from the store... So why not change the type T to JSON to avoid the trap?

Can you give an example of where that would fail ? 
Not sure that I follow your concern.

Reply:
As discussed the objects are simple and thay should be the case.